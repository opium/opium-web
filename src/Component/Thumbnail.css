.Thumbnail {
    position: relative;
    margin: 5px;
    width: 200px;
    height: 200px;
    background-size: cover;
    background-position: center center;
}

.Thumbnail_Caption {
    position: absolute;
    bottom: 0;
    left: 0;
    right: 0;
    padding: 5px;
    background: rgba(89, 79, 63, .6);
    color: #fff;
    text-align: center;
}

.Thumbnail_FolderPlaceholder {
    color: rgb( 89, 79, 63);
}

.Thumbnail_FolderPlaceholder:hover, .Thumbnail_FolderPlaceholder:active {
    color: rgb( 79, 58, 27)
}
