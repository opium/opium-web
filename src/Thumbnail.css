.Thumbnail {
    position: relative;
    margin: 5px;
    height: 200px;
    background-size: cover;
    background-position: center center;
}

.Caption {
    position: absolute;
    bottom: 0;
    left: 0;
    right: 0;
    padding: 5px;
    background: rgba(0, 0, 0, .4);
    color: #fff;
    text-align: center;
}

.Image {
    display: block;
}
