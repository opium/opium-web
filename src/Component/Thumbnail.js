import React, { Component, PropTypes } from 'react';
import Folder from 'react-icons/lib/ti/folder';
import './Thumbnail.css';

class Thumbnail extends Component {
  static propTypes = {
    image: PropTypes.string,
    title: PropTypes.string.isRequired,
    width: PropTypes.number,
    height: PropTypes.number,
  }

  render() {
    const styles = {};
    if (this.props.image) {
      styles.backgroundImage = `url(${this.props.image})`;
    }

    if (this.props.width) {
      styles.width = this.props.width;
    }

    if (this.props.height) {
      styles.height = this.props.height;
    }


    return (
      <article
        className="Thumbnail"
        style={styles}
      >
        {!this.props.image &&
          <Folder
            className="Thumbnail_FolderPlaceholder"
            size={Math.min(this.props.width, this.props.height)}
          />
        }

        <span className="Thumbnail_Caption">
          {this.props.title}
        </span>
      </article>
    );
  }
}

export default Thumbnail;
