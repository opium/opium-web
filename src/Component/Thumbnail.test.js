import React from 'react';
import renderer from 'react-test-renderer';
import Thumbnail from './Thumbnail';

it('renders without crashing', () => {
  const component = renderer.create(
    <Thumbnail
      image="http://demo.opium.sitioweb.fr/2909-vallon-moy-res-jpg/thumbs/200-200"
      title="Alps"
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot()
});
