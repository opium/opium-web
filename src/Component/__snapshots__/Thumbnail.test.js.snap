// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<article
  className="Thumbnail"
  style={Object {}}
>
  <span
    className="Thumbnail_Caption"
  >
    Alps
  </span>
</article>
`;
